Mathlib/Data/Polynomial/Module.lean

```diff
@@ -15,16 +15,73 @@ In this file, we define the polynomial module for an `R`-module `M`, i.e. the `R
 This is defined as a type alias `PolynomialModule R M := ℕ →₀ M`, since there might be different
 module structures on `ℕ →₀ M` of interest. See the docstring of `PolynomialModule` for details.
 
+We also define, given an element `a` in an `R`-algebra `A` and an `A`-module `M`, an `R[X]`-module
+`Module.AEval R M a`, which is a type synonym of `M` with the action of a polynomial `f`
+given by `f • m = Polynomial.aeval a f • m`. In particular `X • m = a • m`.
+
+In the special case that `A = M →ₗ[R] M` and `φ : M →ₗ[R] M`, the module `Module.AEval R M a` is
+abbreviated `Module.AEval' φ`. In this module we have `X • m = ↑φ m`.
 -/
+universe u v
+open Polynomial BigOperators
 
+namespace Module
+/--
+Suppose `a` is an element of an `R`-algebra `A` and `M` is an `A`-module.
+Then `Module.AEval R M a` is the `R[X]`-module with carrier `M`,
+where the action of `f : R[X]` is `f • m = (aeval a f) • m`.
+-/
+@[nolint unusedArguments]
+def AEval (R M : Type*) {A : Type*} [CommSemiring R] [Semiring A] [Algebra R A]
+    [AddCommMonoid M] [Module A M] [Module R M] [IsScalarTower R A M] (_ : A) := M
 
-universe u v
+variable {R A M} [CommSemiring R] [Semiring A] (a : A) [Algebra R A] [AddCommMonoid M] [Module A M]
+  [Module R M] [IsScalarTower R A M]
 
-open Polynomial
+namespace AEval
 
-open Polynomial BigOperators
+instance : AddCommMonoid <| AEval R M a                 := inferInstanceAs (AddCommMonoid M)
+instance : Module R <| AEval R M a                      := inferInstanceAs (Module R M)
+instance : Module A <| AEval R M a                      := inferInstanceAs (Module A M)
```

Comment:
Please give these and the other module instance explicit names, to make it easy to tell them apart in the infoview. Perhaps `instModuleOrig`, `instModuleEval`, and `instModulePolynomial`?

Reply:
Thank you.